homeassistant/components/rest_command/__init__.py

```diff
@@ -58,6 +62,23 @@
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the REST command component."""
 
+    async def reload_service_handler(service: ServiceCall) -> None:
+        """Remove all rest_commands and load new ones from config."""
+        conf = None
+        with contextlib.suppress(HomeAssistantError):
+            conf = await async_integration_yaml_config(hass, DOMAIN)
+        if conf is None or DOMAIN not in conf:
+            return
```

Comment:
If you short circuit out here if there is no configuration, will it remove existing entries if you completely removed the config? It looks like it would leave those intact.

Reply:
The existing config  remains intact in case of failure here. That is similar to the implementations in the `Rest`,  and `HomeKit` components.

Though I doubt the necesity of the suppression here.

Reply:
Just because the same pattern is used in homekit does not mean it's correct.

I agree it's expected that all services are removed if `rest_command` is removed from the configuration.

Reply:
It should remove all the services but not the reload service itself.

The pattern exists in HomeKit and Rest so if you make a mistake, the reload service is not gone and you can still recover.  If its not actually removing the integration instance though, thats a bug

Reply:
Thanks @bdraco 👍 

The pattern in `rest` is  different, we don't exit early if `DOMAIN not in conf`, and there's a specific test testing what happens when the `rest` configuration is removed: `test_reload_and_remove_all` - everything is cleaned up.

What's happening in this PR looks like a bug because the services won't be removed.